platform/ios/src/MGLAnnotationView_Private.h

```diff
@@ -0,0 +1,23 @@
+#import "MGLAnnotationView.h"
+#import "MGLAnnotation.h"
+
+NS_ASSUME_NONNULL_BEGIN
+
+@protocol MGLAnnotationViewDelegate <NSObject>
+
+- (void)annotationView:(MGLAnnotationView *)annotationView didReceiveTapAtPoint:(CGPoint)point;
+
+@end
+
+@interface MGLAnnotationView (Private)
+
+@property (nonatomic, weak) id<MGLAnnotationViewDelegate> delegate;
```

Comment:
I checked and an MKAnnotationView can be added to any arbitrary view. But, I think it is a great idea to remove the constraint of a single delegate. In 984cfb828f9028c2cb9d95e534c8f661cdb1f894 I was able to remove the need for the delegate entirely since your previous suggestion of a hit test in the map view's gesture recognizer is now implemented (and the only need for delegation so far was for the tap interaction). If the need for a delegate comes back for any other reason though, I will go with checking that the containing view is a mapview and casting as such.